client/state/themes/actions.js

```diff
@@ -203,14 +204,24 @@ export function requestThemes( siteId, query = {} ) {
 			query
 		} );
 
+		function filterThemesForJetpack( themes, query ) {
+			return filter( themes, theme => matchThemeByQuery( theme, query ) );
+		}
+
 		return wpcom.undocumented().themes( siteIdToQuery, queryWithApiVersion ).then( ( { found, themes } ) => {
 			dispatch( receiveThemes( themes, siteId ) );
+
+			let filteredThemes = themes;
+			if ( siteIdToQuery ) {
```

Comment:
The check is correct, but I think it's clearer to use the equivalent `siteId !== 'wpcom'` condition -- which more clearly signals, we're dealing with a non-wpcom site.